src/attribute.rs

```diff
@@ -699,8 +715,11 @@ impl<'n, 'f> Iterator for NtfsAttributesRaw<'n, 'f> {
 
         // It's a real attribute.
         let attribute = NtfsAttribute::new(self.file, self.items_range.start, None);
-        self.items_range.start += attribute.attribute_length() as usize;
-
+        let length = usize::try_from(attribute.attribute_length()).ok()?;
```

Comment:
This is fine. However, `usize::try_from` can only fail on platforms where `usize` is shorter than 32-bit (e.g. some embedded platforms). Otherwise, I think the simpler `as usize` should be preferred.

Honestly, the entire code has been written under the assumption that `usize` is at least 32 bits wide.
If that is really a concern, many more parts of the crate need to be rewritten. On the one hand, this affects all parts where `u32` and `usize` are directly converted without error checking. But even more, this affects all parts where we perform slice indexing with 32-bit offsets via `usize`.

As a result, I don't think we should prepare for such platforms within this PR.
You could as well change this back to `as usize` and save a single branch here.

Reply:
Yes I understand, same as the previous conversion, did not know, so just made the least breaking change possible